builtins/amp-ad.js

```diff
@@ -108,8 +108,8 @@ export function installAd(win) {
        */
       this.unlistenViewportChanges_ = null;
 
-      /** @private {boolean} */
-      this.shouldSendIntersectionChanges_ = false;
+      /** @private {IntersectionObserver} */
+      this.intersectionObserver_ = null;
```

Comment:
How would you feel about allowing that to be set late, so you don't have to care about the null case everywhere?